pkg/processor/k8sprocessor/kube/client.go

```diff
@@ -193,11 +193,28 @@ func (c *WatchClient) handlePodUpdate(old, new interface{}) {
 
 func (c *WatchClient) handlePodDelete(obj interface{}) {
 	observability.RecordPodDeleted()
-	if pod, ok := obj.(*api_v1.Pod); ok {
-		c.forgetPod(pod)
-	} else {
+
+	var pod *api_v1.Pod
+
+	switch obj := obj.(type) {
+	case *api_v1.Pod:
+		pod = obj
+	case cache.DeletedFinalStateUnknown:
+		prev, ok := obj.Obj.(*api_v1.Pod)
+		if !ok {
+			c.logger.Error(
+				"object received was DeletedFinalStateUnknown but did not contain api_v1.Pod",
+				zap.Any("received", obj),
+			)
```

Comment:
is it logged often? We would like to avoid extensive error logging if this may be expected behavior

Reply:
If I understand correctly, this is more of a sanity check that shouldn't trigger an error in normal situation, as by design the `cache.DeletedFinalStateUnknown` event should contain the object it pertains to.

Reply:
I needed to double check if this function should always get pod object and confirm that as observer is run per object type